pallets/baby-liminal/src/tests/suite.rs

```diff
@@ -51,44 +67,63 @@ fn does_not_overwrite_registered_key() {
         put_key();
 
         assert_err!(
-            BabyLiminal::store_key(caller(), IDENTIFIER, vk()),
+            BabyLiminal::store_key(owner(), IDENTIFIER, vk()),
             Error::<TestRuntime>::IdentifierAlreadyInUse
         );
     });
 }
 
 #[test]
-fn caller_cannot_delete_key() {
+fn not_owner_cannot_delete_key() {
     new_test_ext().execute_with(|| {
         put_key();
-        assert_err!(BabyLiminal::delete_key(caller(), IDENTIFIER), BadOrigin);
+        assert_err!(
+            BabyLiminal::delete_key(not_owner(), IDENTIFIER),
+            Error::<TestRuntime>::NotOwner
+        );
     });
 }
 
 #[test]
-fn sudo_can_delete_key() {
+fn owner_can_delete_key() {
     new_test_ext().execute_with(|| {
         put_key();
-        assert_ok!(BabyLiminal::delete_key(root(), IDENTIFIER));
+        assert_ok!(BabyLiminal::delete_key(owner(), IDENTIFIER));
     });
 }
 
 #[test]
-fn caller_cannot_overwrite_key() {
+fn not_owner_cannot_overwrite_key() {
     new_test_ext().execute_with(|| {
         put_key();
         assert_err!(
-            BabyLiminal::overwrite_key(caller(), IDENTIFIER, vk()),
-            BadOrigin
+            BabyLiminal::overwrite_key(not_owner(), IDENTIFIER, vk()),
+            Error::<TestRuntime>::NotOwner
         );
     });
 }
 
 #[test]
-fn sudo_can_overwrite_key() {
+fn owner_can_overwrite_key() {
     new_test_ext().execute_with(|| {
         put_key();
-        assert_ok!(BabyLiminal::overwrite_key(root(), IDENTIFIER, vk()));
+        assert_ok!(BabyLiminal::overwrite_key(owner(), IDENTIFIER, vk()));
+    });
+}
+
+#[test]
+fn key_deposits() {
+    new_test_ext().execute_with(|| {
+        let reserved_balance_before = reserved_balance(1);
+        let deposit = put_key();
+        let reserved_balance_after = reserved_balance(1);
+
+        assert_eq!(reserved_balance_after - reserved_balance_before, deposit);
+        assert_ok!(BabyLiminal::overwrite_key(owner(), IDENTIFIER, vk()));
+
+        assert_eq!(reserved_balance_after, reserved_balance(1));
+        assert_ok!(BabyLiminal::delete_key(owner(), IDENTIFIER));
+        assert_eq!(reserved_balance_before, reserved_balance(1));
```

Comment:
We're still missing the test for overwriting with a key of different length.

Reply:
added